visualizer.py

```diff
@@ -0,0 +1,323 @@
+        except ValueError:
+            self._capture_duration_input.value = str(float(old_value))
+
+
+    def update(self):
+        """ Periodic check for capture status. """
+        trigger = self.receiver.trigger
+        self._trigger_state.text = f'Trigger State: <b>{trigger.name.upper()}</b>'
+
+        if trigger is TriggerState.STOPPED:
+            times, traces = self.receiver.get_traces()
+            self._redraw(times=times, traces=traces)
+
+            # Disable the document callback now that we have redrawn the plot.
+            if self._callback is not None:
+                self.doc.remove_periodic_callback(self._callback)
+                self._callback = None
+
+
+    def _redraw(self, times: List[float], traces: Mapping[str, Trace]):
+        """ Redraw plots.
+
+        # Args
+            times: The list of timestamps of trace points.
+            traces: A dictionary of all traces to display.
+        """
+        # Update the data store atomically
+        new_datastore = {
+            'time': times / SAMPLE_RATE_HZ,
+        }
+
+        for trace in traces.values():
+            new_datastore[trace.label] = trace.values * trace.scale
+
+        self._data_store = new_datastore
+
+        figure = bokeh.plotting.figure(output_backend="webgl", sizing_mode="stretch_both")
+
+        # Update traces
+        palette = bokeh.palettes.d3['Category10'][len(traces.keys())]
+        for (trace, color) in zip(traces.values(), palette):
+            figure.circle(x='time', y=trace.label, source=self._data_store, color=color,
+                          legend_label=trace.label)
+
+        figure.legend.location = 'top_left'
+        figure.legend.click_policy = 'hide'
+
+        self.layout.children[0] = figure
+
+
+    def capture(self, duration: float = None):
+        """ Initialize a data capture from Stabilizer's datastream.
+
+        Args:
+            duration: The duration to capture for.
+        """
+        if duration is None:
+            duration = float(self._capture_duration_input.value)
+
+        self.receiver.capture(duration)
+
+        # TODO: Force the trigger for now. Once we allow triggered waveforms, this force will be
+        # removed.
+        self.receiver.force_trigger()
+
+        # Start periodicly reading trigger state.
+        self._callback = self.doc.add_periodic_callback(self.update,
+                                                        period_milliseconds=1000/60)
+
+
+async def configure_streaming(prefix, broker, port):
+    """ Set up Stabilizer streamining to the desired endpoint.
+
+    Args:
+        port: The port to direct the stream to.
+        broker: The IP address of the broker.
+        prefix: The miniconf prefix of Stabilizer.
+    """
+    _, local_ip = _get_ip(broker)
+    interface = await Miniconf.create(prefix, broker)
+
+    # Configure the stream
+    print(f'Configuring stream to target {".".join(map(str, local_ip))}:{port}')
```

Comment:
`logger.debug()` at most. Less code and less formatting is better here. It's at most debugging information for the developer.